bytecomp/bytegen.ml

```diff
@@ -1239,20 +1239,23 @@ let reset () =
   compunit_name := "";
   Stack.clear functions_to_compile
 
-let compile_implementation modulename expr =
+let compile_gen ?modulename ~init_stack expr =
   reset ();
-  compunit_name := modulename;
+  begin match modulename with
+  | Some name -> compunit_name := name
+  | None -> ()
```

Comment:
Not introduced by this PR, but this code means that compunit_name will be left to its previous value. To me it would be better reset the ref to a default when modulename is not set

Reply:
The call to `reset` one line above should set it to its default value `""`.